ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/PushQueryValidator.java

```diff
@@ -17,17 +17,12 @@
 
 import io.confluent.ksql.analyzer.Analysis.AliasedDataSource;
 import io.confluent.ksql.metastore.model.DataSource.DataSourceType;
-import io.confluent.ksql.parser.tree.ResultMaterialization;
 import io.confluent.ksql.util.KsqlException;
 
 public class PushQueryValidator implements QueryValidator {
 
   @Override
   public void validate(final Analysis analysis) {
-    if (analysis.getResultMaterialization() != ResultMaterialization.CHANGES) {
```

Comment:
Isn't the first part of the if condition redundant though? Can we simplify this to just `if (analysis.getResultMaterialization().get() == ResultMaterialization.FINAL)`

Reply:
if `getResultMaterialization.isPresent()` returns `false`, then `analysis.getResultMaterialization().get()` will throw a `NoSuchElementException`

```java
    /**
     * If a value is present in this {@code Optional}, returns the value,
     * otherwise throws {@code NoSuchElementException}.
     *
     * @return the non-null value held by this {@code Optional}
     * @throws NoSuchElementException if there is no value present
     *
     * @see Optional#isPresent()
     */
    public T get() {
        if (value == null) {
            throw new NoSuchElementException("No value present");
        }
        return value;
    }
```